### PR TITLE
[ios] bump deployment target to 9.0 & use newer Xcode on jenkins

### PR DESCRIPTION
**Pull request by Rechi**
## Description
- bump ios deployment target to 9.0
- correct some osx documentation
- use Xcode 9.0 on jenkins for ios and osx

## Motivation and Context
#13454 requires iOS 9 as deployment version

## Checklist:
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by jimfcarroll** [comment]
My PR #13454 already has the bump and everything passes.

**Comment by Memphiz** [comment]
@rechi - is looked at it already and made line comments

**Comment by Memphiz** [comment]
Mhhh i think this is some sort of mis communication - I surely wanted some answers to my in line questions - can‘t you see those for some reason?

**Comment by Rechi** [comment]
I don't see any here and also none in #13454.
Where should I see them?

**Comment by Memphiz** [comment]
Damn was not aware that I need to „submit“ the review - what a stupid UI

File: docs/README.ios.md [hunk 1]
```diff
@@ -54,10 +52,6 @@ it either from the macOS AppStore (Xcode) or from the Apple Developer Homepage.
 As far as we know the compilation for iOS should work with the following
 constellations of Xcode and macOS versions (to be updated once we know more):
 
-1. XCode 6.0.1 against iOS SDK 8.0 on 10.9 (Mavericks)
-2. XCode 6.1.0 against iOS SDK 8.1 on 10.10 (Yosemite)
-3. XCode 6.3.0 against iOS SDK 8.3 on 10.10 (Yosemite)
-4. Xcode 6.4.0 against iOS SDK 8.4 on 10.10 (Yosemite)
 5. Xcode 7.x against iOS SDK 9.x on 10.10 (Yosemite)
```
[the remark this answers is not in the document]
Comment (Rechi) [reply docs/README.ios.md]: this is only a deployment target bump, and Xcode 7 has iOS SDK 9

File: docs/README.osx.md [hunk 2]
```diff
@@ -29,17 +29,14 @@ There are 3 ways to build Kodi for Mac:
 Kodi for Mac is composed of a main binary with numerous dynamic libraries and
 codecs that support a multitude of music and video formats.
 
-The minimum version of OSX you need to run(!) Kodi is 10.7 atm.
+The minimum version of OSX you need to run(!) Kodi is 10.8 atm.
 
 - On Mavericks (OSX 10.9.x) we recommend using Xcode 6.1.
 - On Yosemite (OSX 10.10.x) we recommend using Xcode 6.4.
 - On El Capitan (OSX 10.11.x) we recommend using Xcode 7.x or Xcode 8.x.
```
[the remark this answers is not in the document]
Comment (Rechi) [reply docs/README.osx.md]: same

File: docs/README.osx.md [hunk 3]
```diff
@@ -29,17 +29,14 @@ There are 3 ways to build Kodi for Mac:
...
 - On Sierra (macOS 10.12.x) we recommend using Xcode 8.x.
 - On High Sierra (macOS 10.13.x) we recommend using Xcode 9.x.
 
-**ATTENTION**: You need to download and install XQuartz from <https://xquartz.macosforge.org/landing/>
-since its not part of macOS anymore since 10.8.
-
```
[the remark this answers is not in the document]
Comment (Rechi) [reply docs/README.osx.md]: was it needed for a lib or for something in core Kodi code?

File: tools/depends/target/config.site.in [hunk 4]
```diff
@@ -159,7 +159,7 @@ if test "@platform_os@" = "ios"; then
   # tweaks for libffi (ios must use llvm-gcc-4.2)
   if test "${PACKAGE_NAME}" = "libffi" ; then
       case "@use_xcode@" in
-        3.*.* | 4.* | 4.*.*)
+        4.* | 4.*.*)
```
[the remark this answers is not in the document]
Comment (Rechi) [reply tools/depends/target/config.site.in]: will do some more cleanup later

File: docs/README.osx.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Memphiz) [reply docs/README.osx.md]: Not sure - must have been a x11 header somewhere which was not used on Linux platform only - I always only noticed it on fresh Mac OS installs obviously - leave it out for now in the docs..

**Pull request merged**